include/boost/url/url_base.hpp

```diff
@@ -1659,6 +1674,34 @@ class url_base::segments_type
     BOOST_URL_DECL
     iterator
     end() const noexcept;
+
+    BOOST_URL_DECL
+    iterator
+    erase( iterator b, iterator e ) noexcept;
+
+    BOOST_URL_DECL
+    iterator
+    erase( iterator at ) noexcept;
+
+    // Strong guarantee.
+    BOOST_URL_DECL
+    iterator
+    insert_encoded_segment( iterator pos, string_view s );
+
+    // Strong guarantee.
+    BOOST_URL_DECL
+    iterator
+    insert_segment( iterator pos, string_view s );
+
+    // Strong guarantee.
+    BOOST_URL_DECL
+    iterator
+    replace_encoded_segment( iterator pos, string_view s );
+
+    // Strong guarantee.
+    BOOST_URL_DECL
+    iterator
+    replace_segment( iterator pos, string_view s );
```

Comment:
All these functions need javadocs. The best time to write the reference documentation is when you implement the function since it will be fresh in your head. Coming back to it weeks or months later and you might forget things.

Reply:
Yes all functions need to be documented. I prefer if you approve the interface before I write the documentation so I don't have to document things that get rejected.